apps/studio/components/interfaces/DiskManagement/DiskManagement.schema.ts

```diff
@@ -0,0 +1,169 @@
+          path: ['totalSize'],
+        })
+      }
+    }
+
+    if (storageType === 'gp3') {
+      const maxIopsAllowedForDiskSizeWithGp3 = calculateMaxIopsAllowedForDiskSizeWithGp3(totalSize)
+
+      if (provisionedIOPS > IOPS_RANGE[DiskType.GP3].max) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `IOPS can not exceed ${formatNumber(IOPS_RANGE[DiskType.GP3].max)} for GP3 Disk. Change the Disk type to io2 for higher IOPS support.`,
+          path: ['provisionedIOPS'],
+        })
+      }
+
+      if (provisionedIOPS < IOPS_RANGE[DiskType.GP3].min) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `IOPS must be at least ${formatNumber(IOPS_RANGE[DiskType.GP3].min)}`,
+          path: ['provisionedIOPS'],
+        })
+      } else if (provisionedIOPS > maxIopsAllowedForDiskSizeWithGp3) {
+        if (totalSize >= 8) {
+          const diskSizeRequiredForIopsWithGp3 =
+            calculateDiskSizeRequiredForIopsWithGp3(provisionedIOPS)
+
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Larger Disk size of at least ${formatNumber(diskSizeRequiredForIopsWithGp3)} GB required. Current max is ${formatNumber(maxIopsAllowedForDiskSizeWithGp3)} IOPS.`,
+            path: ['provisionedIOPS'],
+          })
+        } else {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Invalid IOPS value due to invalid disk size`,
+            path: ['provisionedIOPS'],
+          })
+        }
+      }
+
+      if (throughput !== undefined) {
+        const maxThroughput = Math.min(0.25 * provisionedIOPS, 1000)
+
+        if (throughput > THROUGHPUT_RANGE['gp3'].max) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Throughput can not exceed ${formatNumber(maxThroughput)} MiB/s`,
+            path: ['throughput'],
+          })
+        }
+        if (throughput > maxThroughput) {
+          const iopsRequiredForThroughput = calculateIopsRequiredForThroughput(throughput)
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Need at least ${formatNumber(iopsRequiredForThroughput)} IOPS to support ${formatNumber(throughput)} MiB/s.`,
+            path: ['throughput'],
+          })
+        }
+        if (throughput < THROUGHPUT_RANGE[DiskType.GP3].min) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Throughput must be at least ${formatNumber(THROUGHPUT_RANGE[DiskType.GP3].min)} MiB/s`,
+            path: ['throughput'],
+          })
+        }
+      }
+
+      if (totalSize > DISK_LIMITS[DiskType.GP3].maxStorage) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: `Allocated disksize must not exceed ${formatNumber(DISK_LIMITS[DiskType.GP3].maxStorage)} GB`,
```

Comment:
nit: spelling consistency

```suggestion
          message: `Allocated disk size must not exceed ${formatNumber(DISK_LIMITS[DiskType.GP3].maxStorage)} GB`,
```